### DISCUSSION TITLE
xxx-cp38-cp38-macosx_12_4_arm64.whl is not a supported wheel on this platform

### DISCUSSION THREAD

**Discussion by hombit** [pypa/cibuildwheel#1168] [Q&A] [answered]
I'm trying to use `cibuildwheel` locally on my ARM mac (macOS 12.4) for building [`light-curve`](https://github.com/light-curve/light-curve-python) package. I followed #937 thread and found that my Rust installation is `minos 11.0` and my only dynamically linked library is `minos 12` so I set target to 12.0:
```sh
CIBW_ENVIRONMENT='MACOSX_DEPLOYMENT_TARGET=12.0' CIBW_BEFORE_ALL='' python3 -mcibuildwheel --platform macos
```
However I still have the non-supported wheel error
```
+ pip install /private/var/folders/w1/lh3h4s7d5g10rdlfj4h0mshw0000gn/T/cibw-run-o1kp1gsl/cp38-macosx_arm64/repaired_wheel/light_curve-0.6.0-cp38-cp38-macosx_12_0_arm64.whl
ERROR: light_curve-0.6.0-cp38-cp38-macosx_12_0_arm64.whl is not a supported wheel on this platform.
```

Could you please point me to any possible root of this problem?

**Comment by Czaki** [comment, marked as answer]
could you check if you run native python or x86 one that uses the Rosetta emulation layer?

If I good remember this should be enough:

```
python -c "import platform; print(platform.machine())"
```

But I do not have access to macOS m1 machine for verification.

**Reply by hombit** [reply]
Thanks for the idea! I run cibuildwheel via python 3.9 installed via brew, it is ARM python
```
python3 -c "import platform; print(platform.machine())"
arm64
```
And I believe cibuildwheel builds and tests everything using python 3.8 installed by the package from python.org, as suggested by the manual.

**Reply by hombit** [reply]
Ah, you are right, sorry! Official python package installed x86_64 python!
```
/Library/Frameworks/Python.framework/Versions/3.8/bin/python3.8 -c "import platform; print(platform.machine())"
x86_64
```

**Reply by joerick** [reply]
It would be worth checking `/Library/Frameworks/Python.framework/Versions/3.8/bin/python3.8 -c "import platform; print(platform.machine())"` too, as you might have an older version of Python 3.8 installed that didn't support arm64.

edit: you beat me by 1 minute!

**Reply by Czaki** [reply]
Hm. But You told me that you use python 3.9 and try to install the wheel for python 3.8. It will not work. You need to use the wheel for python 3.9 on python 3.9.

**Reply by joerick** [reply]
The version that cibuildwheel uses is https://www.python.org/ftp/python/3.8.10/python-3.8.10-macosx10.9.pkg 

https://github.com/pypa/cibuildwheel/blob/main/cibuildwheel/resources/build-platforms.toml#L80

**Reply by hombit** [reply]
Sorry, @Czaki, I mean I run cibuildwheel using python3.9, but it uses python3.8 to build, test, etc.

**Reply by hombit** [reply]
@joerick could be a bug in cibuildwheel config, because this package is not universal2, it is x86_64. Probably it is what caused my problem. I installed https://www.python.org/ftp/python/3.8.10/python-3.8.10-macos11.pkg and it works now

**Reply by hombit** [reply]
@joerick you see, the link is the same as for x86_64 python3.8 write universal2 python3.9 and python3.10 look correct

**Reply by hombit** [reply]
Created a PR
https://github.com/pypa/cibuildwheel/pull/1169

**Reply by joerick** [reply]
Ah yes you are right. But that PR is probably not going to work, I'll respond in there.
